octopoes/octopoes/api/router.py

```diff
@@ -514,8 +515,10 @@ def delete_node(xtdb_session_: XTDBSession = Depends(xtdb_session)) -> None:
         raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail="Deleting node failed") from e
 
 
-@router.post("/bits/recalculate", tags=["Bits"])
-def recalculate_bits(octopoes: OctopoesService = Depends(octopoes_service)) -> int:
+RECALCULATE_BITS_SESSIONS: dict[uuid.UUID, int] = {}
```

Comment:
Although simple, there is an issue with this approach: what if clients configure `uvicorn` to run with more than 1 worker, each worker has its own state. Hence we wouldn't have read-your-writes consistency, and you randomly see or don't see the sessions you created, depending on which Uvicorn worker handles the request.

Reply:
Right the other approach would be to dump the data in XTDB, @underdarknl and I talked about that scenario. 

1. Do clients do this?
1. Does a user "switch" worker on every request?

Reply:
I did a quick check using this code:

```python3
import os
import uvicorn
from contextlib import asynccontextmanager

from fastapi import FastAPI

SESSIONS = []

@asynccontextmanager
async def lifespan(app: FastAPI):
    SESSIONS.append(os.getpid())
    yield

app = FastAPI(lifespan=lifespan)

@app.get("/")
async def index():
    return {"worker_id": os.getpid(), "sessions": SESSIONS}

if __name__ == "__main__":
    uvicorn.run("main:app", host="127.0.0.1", port=8009, workers=2)
```

And get different responses quite randomly:
![image](https://github.com/user-attachments/assets/213ac428-798c-4ed9-8503-c820d9542fac)

So I'd answer "yes" on the second question. Is your first question: do clients use multiple workers?

Reply:
I think they would have to if they want to scale OpenKAT, because I doubt a single octopoes process is going to be enough for a big installation. And if you would run OpenKAT on Kubernetes, you would normally also create a Deployment with multiple replicas. In both cases there is no guarantee that a subsequent request goes to the same process.

The problem is that we are trying to run a task inside a http request for which it is always better to use task runner. We already have celery, so in my opinion the right approach would be to create a celery task for recalculating bits. I can help with that if needed, I have a lot of experience with celery.

Reply:
Thanks @Donnype that proves you point quite elegantly. If @dekkers and @underdarknl agree I will rewrite this for the using XTDB as central storage -- with the all important GET. Thanks for the feedback!